src/dag-node/util.js

```diff
@@ -0,0 +1,36 @@
+'use strict'
+
+const DAGLink = require('./../dag-link')
+
+exports = module.exports
+
+function cloneData (dagNode) {
+  let data = new Buffer(0)
+
+  if (dagNode.data && dagNode.data.length > 0) {
+    data = new Buffer(dagNode.data.length)
+    dagNode.data.copy(data)
+  }
+
+  return data
+}
+
+function cloneLinks (dagNode) {
+  return dagNode.links.length > 0 ? dagNode.links.slice() : []
+}
+
+function linkSort (a, b) {
+  return (new Buffer(a.name || '', 'ascii').compare(new Buffer(b.name || '', 'ascii')))
```

Comment:
This is super wasteful, the link names should be converted to buffers once and then sorted, not on  every sorting call.